InstantSearch.js/routing-seo-friendly/src/search-router.js

```diff
@@ -0,0 +1,86 @@
+/* global instantsearch */
+
+function getCategorySlug(name) {
+  return name.replace(/\s/g, '-');
+}
+
+function getCategoryName(slug) {
+  return slug.replace(/-/g, ' ');
+}
+
+const router = instantsearch.routers.history({
+  windowTitle({ category, q }) {
+    const query = q ? `Results for "${q}"` : 'Search';
+
+    return [category, query].filter(Boolean).join(' – ');
```

Comment:
while smart, I had to read it twice to understand what it was doing, maybe just an if would make sense?

Reply:
It makes more sense to join them here. We could have more parts in the title. Besides, an implementation with an `if` would be more difficult to understand IMO because the `–` dash shouldn't be displayed if one of the items are not defined.

Reply:
I mean:

```js
if (category) {
  return  [category, query].join(' – ');
}
return query
```